Fixes after rebase to Django 1.8

symbolist opened the pull request:
https://openedx.atlassian.net/browse/TNL-2506

symbolist commented:
@muzaffaryousaf @muhammad-ammar There are a few errors related to the message view. Please ignore them for now. I am working on fixing them.

muzaffaryousaf commented on lms/static/js/bookmarks/collections/bookmarks.js:
```diff
@@ -0,0 +1,24 @@
+            });
+    });
+
+})(define || RequireJS.define);
```
New line.

muzaffaryousaf commented on common/test/acceptance/tests/lms/test_lms_courseware.py:
```diff
@@ -67,6 +68,18 @@ def _goto_problem_page(self):
+        """
+        self.course_outline.q(css=".subsection-header-actions .configure-button").first.click()
+        self.course_outline.q(css="#start_date").fill("01/01/2030")
+        self.course_outline.q(css=".action-save").first.click()
```
How we ended up with this method ? I don't remember this method as part of bookmarks epic.

symbolist replied:
I have removed it.

muzaffaryousaf commented on common/test/acceptance/tests/lms/test_lms_courseware.py:
```diff
@@ -92,6 +105,9 @@ def test_courseware(self):
 
+        # Wait for 2 seconds to save new date.
+        time.sleep(2)
+
```
Explicit wait ? is it by intention ?

symbolist replied:
Got in during the rebase. I have removed it.

muzaffaryousaf commented on lms/djangoapps/courseware/module_render.py:
```diff
@@ -743,6 +744,8 @@ def rebind_noauth_module_to_user(module, real_user):
             "reverification": ReverificationService(),
             'proctoring': ProctoringService(),
             'credit': CreditService(),
+            'reverification': ReverificationService(),
```
Duplicate registration. See the line 744.

symbolist replied:
Removed.

muzaffaryousaf commented on lms/envs/common.py:
```diff
@@ -1228,6 +1229,7 @@
         for pth in ['courseware', 'histogram', 'navigation']
     ] +
     ['js/' + pth + '.js' for pth in ['ajax-error']] +
+    ['js/bookmarks/main.js'] +
```
Are we gonna do js optimization for bookmarks, courseware search feature is being changed and probably we need to change this into factory format as well. `ad16f9f968fd2a0d8a144759c88d8d45d60e38d2`

symbolist replied:
Yes, this was removed when we changed to using require.js optimizer.

muzaffaryousaf commented on lms/static/js/spec/main.js:
```diff
@@ -86,6 +86,13 @@
+            'js/bookmarks/views/bookmarks_list_button': 'js/bookmarks/views/bookmarks_list_button',
+            'js/bookmarks/views/bookmarks_list': 'js/bookmarks/views/bookmarks_list',
+            'js/bookmarks/views/bookmark_button': 'js/bookmarks/views/bookmark_button',
+            'js/views/message_banner': 'js/views/message_banner',
```
Earlier it was `'js/views/message'`, is't it ?

muzaffaryousaf replied:
oh, Did we decided to call message_banner instead of message ?

symbolist replied:
It was initially `message_banner`. Since it has since then been used in other places too, I have decided to retain the original name. For some reason, the spec file was not included here.

muzaffaryousaf commented on lms/static/js/bookmarks/views/bookmarks_list.js:
```diff
@@ -0,0 +1,129 @@
+            'common/js/components/views/paging_header', 'common/js/components/views/paging_footer'],
+        function (gettext, $, _, Backbone, Logger, _moment, PagingHeaderView, PagingFooterView) {
+
+        var moment = _moment || window.moment;
```
I don't see moment_locale added anywhere in settings/configurations.

muzaffaryousaf commented on lms/static/js/spec/student_profile/learner_profile_fields_spec.js:
```diff
@@ -5,7 +5,7 @@ define(['backbone', 'jquery', 'underscore', 'common/js/spec_helpers/ajax_helpers
        ],
     function (Backbone, $, _, AjaxHelpers, TemplateHelpers, Helpers, UserAccountModel, LearnerProfileFields,
-              MessageBannerView) {
+              MessageView) {
```
Should't it be `MessageBannerView` ?

muzaffaryousaf commented on lms/static/js/spec/student_profile/learner_profile_view_spec.js:
```diff
@@ -11,7 +11,7 @@ define(['backbone', 'jquery', 'underscore', 'common/js/spec_helpers/ajax_helpers
     function (Backbone, $, _, AjaxHelpers, TemplateHelpers, Helpers, LearnerProfileHelpers, FieldViews,
               UserAccountModel, AccountPreferencesModel, LearnerProfileFields, LearnerProfileView,
-              AccountSettingsFieldViews, MessageBannerView) {
+              AccountSettingsFieldViews, MessageView) {
```
`MessageBannerView` ?

muzaffaryousaf commented on lms/static/js/spec/views/message_banner_spec.js:
```diff
@@ -1,27 +1,46 @@
+            };
+
+            it('renders correctly with the /fields/message_banner template', function() {
+                var messageView = createMessageView(messageSelector, '#message_banner-tpl');
```
`messageSelector` is no more defined.

muzaffaryousaf replied:
Quality is failing for this too.

muzaffaryousaf commented on lms/static/js/spec/views/message_banner_spec.js:
```diff
@@ -1,27 +1,46 @@
+define(['backbone', 'jquery', 'underscore', 'js/views/message_banner', 'js/common_helpers/template_helpers'
        ],
-    function (Backbone, $, _, MessageBannerView) {
+    function (Backbone, $, _, MessageView, TemplateHelpers) {
```
Can we change all `MessageView` to `MessageBannerView` for consistency.

symbolist replied:
I have changed all instances of `MessageView` to `MessageBannerView`.

muzaffaryousaf commented on lms/static/js/views/message_banner.js:
```diff
@@ -37,4 +37,4 @@
         return MessageBannerView;
     });
-}).call(this, define || RequireJS.define);
+}).call(this, define || RequireJS.define);
```
New line.

muzaffaryousaf replied:
We did some changes in message_banner which i'm not able to see in this file. You can see them here https://github.com/edx/edx-platform/pull/8082/files#diff-db57c2f627cd8ff4221f70cb6dbac64bR9

symbolist replied:
I have reverted those changes since some other changes were made to `message_banner`. This change only added a different template but the original template has since been enhanced to support icons.

muzaffaryousaf commented on lms/static/sass/search/_search.scss:
```diff
@@ -124,7 +123,7 @@
 
 .courseware-search-bar {
-  box-shadow: 0 1px 0 $white inset, 0 -1px 0 $shadow-l1 inset;
+   width: flex-grid(7);
```
Not able to see these changes https://github.com/edx/edx-platform/pull/8082/files#diff-2ddc5e8947efe274084196d8b90b41b2R221

symbolist replied:
I don't think that sass is needed anymore.

muzaffaryousaf commented on openedx/core/djangoapps/credit/serializers.py:
```diff
@@ -6,33 +6,18 @@
-        except InvalidKeyError as ex:
-            raise serializers.ValidationError("Invalid course key: {msg}".format(msg=ex.msg))
-
-
```
This is not part of bookmarks, is it?

symbolist replied:
It is being used in the serializer now. I have moved it outside the `credit` app.

muzaffaryousaf commented:
@symbolist These changes are missing in this PR https://github.com/edx/edx-platform/pull/8082/files#diff-7a56670e4a95d6bc8aae2fb88d8074bbR15

muhammad-ammar commented:
@symbolist currently sass is broken on courseware page, are we getting some help from frrances?

muhammad-ammar commented:
@symbolist other than sass, things are looking good to me.

symbolist commented:
@muzaffaryousaf I have added the missing changes from `user_api/urls.py` and addressed the rest of your feedback. 

@muhammad-ammar Yes, the UX team is working on refining the styling.

muhammad-ammar commented:
Except sass, lgtm :+1:

muzaffaryousaf commented:
:+1: LGTM except SASS.

muzaffaryousaf commented:
Don't forget to squash the commits.
